### DISCUSSION TITLE
Is it by design that Prompt Section's `renderAsText` and `renderAsMessage` will return result with different length for same content?

### DISCUSSION THREAD

**Discussion by blackchoey** [microsoft/teams-sdk#1226] [Q&A]
Take following conversation history as example:
```
[
    { role: 'user', content: 'Hello' },
    { role: 'assistant', content: 'Hi! How can I help you?' },
    { role: 'user', content: "I'd like to book a flight" },
    { role: 'assistant', content: 'Sure, where would you like to go?' }
]
```
The `renderAsText` function will return result with token length 36, but `renderAsMessage` will return result with token length 25.
Seems the major difference is `renderAsMessage` does not count the length of prefixes (like "user: ", etc.). But since the message will be rendered as text finally, items from `renderAsMessage` may be truncated again if the length is approaching the limit. I'm not sure whether this will cause possible confusions.

**Comment by corinagum** [comment]
This seems like a bug to me. Ideally having notes to explain differences like this in our documentation would help prevent confusion in the future.

**Reply by corinagum** [reply]
@Stevenic Do you have any input?
